provider/azure/credentials.go

```diff
@@ -46,5 +84,26 @@ func (environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, e
 
 // FinalizeCredential is part of the environs.ProviderCredentials interface.
 func (environProviderCredentials) FinalizeCredential(ctx environs.FinalizeCredentialContext, in cloud.Credential) (cloud.Credential, error) {
-	return in, nil
+	switch authType := in.AuthType(); authType {
+	case cloud.UserPassAuthType:
+		fmt.Fprintf(ctx.GetStderr(), `
+WARNING: The %q auth-type is deprecated, and will be removed soon.
+
+Please update the credential in ~/.local/share/juju/credentials.yaml,
+changing auth-type to %q, and dropping the tenant-id field.
+
+`[1:],
+			authType, clientCredentialsAuthType,
+		)
+		attrs := in.Attributes()
+		delete(attrs, credAttrTenantId)
+		label := in.Label
+		in = cloud.NewCredential(clientCredentialsAuthType, attrs)
+		in.Label = label
+		fallthrough
```

Comment:
Yeah, I was thinking we might need to do something for the resultant auth-type. I don't think we ever will, so updated.